samples/boards/nrf52/mesh/onoff_level_app/src/mesh/device_composition.c

```diff
@@ -0,0 +1,340 @@
+		printk("Unable to send LEVEL_SRV Status response\n");
+	}
+}
+
+static void gen_level_set_unack(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	int err;
+	struct net_buf_simple *msg = model->pub->msg;
+	struct server_state *state = model->user_data;
+
+	state->current = net_buf_simple_pull_le16(buf);
+
+	if (state->previous == state->current) {
+		return;
+	}
+
+	if (state->model_instance == 0x01) {
+		light_state_current.power = state->current;
+		update_light_state();
+	}
+
+	if (model->pub->addr != BT_MESH_ADDR_UNASSIGNED) {
+
+		bt_mesh_model_msg_init(msg,
+				       BT_MESH_MODEL_OP_2(0x82, 0x08));
+
+		net_buf_simple_add_le16(msg, state->current);
+
+		err = bt_mesh_model_publish(model);
+
+		if (err) {
+			printk("bt_mesh_model_publish err %d\n", err);
+		}
+	}
+
+	state->previous = state->current;
+}
+
+static void gen_level_set(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	gen_level_set_unack(model, ctx, buf);
+	gen_level_get(model, ctx, buf);
+}
+
+static void gen_delta_set_unack(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	int err;
+	struct net_buf_simple *msg = model->pub->msg;
+	struct server_state *state = model->user_data;
+
+	s32_t tmp32 = state->current + net_buf_simple_pull_le32(buf);
+	uint8_t tid = net_buf_simple_pull_u8(buf);
+
+	if (state->last_tid == tid && state->last_tx_addr == ctx->addr) {
+		return;
+	}
+
+	state->last_tid = tid;
+	state->last_tx_addr = ctx->addr;
+
+	if (tmp32 < -32768) {
+		tmp32 = -32768;
+
+	} else if (tmp32 > 32767) {
+		tmp32 = 32767;
+	}
+
+	state->current = (int16_t)tmp32;
+
+	printk("Level -> %d\n", state->current);
+
+	if (state->model_instance == 0x01) {
+		light_state_current.power = state->current;
+		update_light_state();
+	}
+
+	if (model->pub->addr != BT_MESH_ADDR_UNASSIGNED) {
+
+		bt_mesh_model_msg_init(msg,
+				       BT_MESH_MODEL_OP_2(0x82, 0x08));
+
+		net_buf_simple_add_le16(msg, state->current);
+
+		err = bt_mesh_model_publish(model);
+
+		if (err) {
+			printk("bt_mesh_model_publish err %d\n", err);
+		}
+	}
+
+	state->previous = state->current;
+}
+
+static void gen_delta_set(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	gen_delta_set_unack(model, ctx, buf);
+	gen_level_get(model, ctx, buf);
+}
+
+static void gen_move_set_unack(struct bt_mesh_model *model,
+			       struct bt_mesh_msg_ctx *ctx,
+			       struct net_buf_simple *buf)
+{
+}
+
+static void gen_move_set(struct bt_mesh_model *model,
+			 struct bt_mesh_msg_ctx *ctx,
+			 struct net_buf_simple *buf)
+{
+	gen_move_set_unack(model, ctx, buf);
+	gen_level_get(model, ctx, buf);
+}
+
+static void gen_level_cli_status(struct bt_mesh_model *model,
+				 struct bt_mesh_msg_ctx *ctx,
+				 struct net_buf_simple *buf)
+{
+	s16_t tmp16 = net_buf_simple_pull_le16(buf);
```

Comment:
There should always be an empty line after variable declarations, however in this case you could just remove the variable and do the call to net_buf_simple_pull_le16() from the parameters of the printk() call.